### ISSUE TITLE
Show complete response in case of an error in HttpJsonClient

### ISSUE THREAD

**Issue by fthomas** [fthomas/scala-steward#668]
`HttpJsonClient.request` currently uses `expect` to a make a request. If the request fails, only the status code is returned which is not very helpful for debugging. We should use `expectOr` here and log the full response instead.

**Comment by BaraiKaran** [comment]
I would like to contribute to resolving this issue.

**Comment by fthomas** [comment]
@BaraiKaran Thanks for your interest in working on this issue. My idea is to replace `expect` with `expectOr` like this:
```diff
   private def request[A: Decoder](method: Method, uri: Uri, modify: ModReq): F[A] =
-    client.expect[A](modify(Request[F](method, uri)))(jsonOf[F, A])
+    client.expectOr[A](modify(Request[F](method, uri)))(response => ???)(jsonOf[F, A])
```
`???` there should be a `F[Throwable]` which contains the complete headers and body of `response` in its `getMessage`. Http4s uses [`UnexpectedStatus`](https://github.com/http4s/http4s/blob/59bcf181e74e1c05bd24c68847391772369fffe2/client/src/main/scala/org/http4s/client/Client.scala#L251) in case of `expect` and I think we should have a new class (`UnexpectedResponse`) which would contain the complete response.

If you have any questions, please feel free to ask.

**Comment by robstoll** [comment]
@BaraiKaran are you working on this one? I have a WIP which you could use if you haven't started yet

**Comment by BaraiKaran** [comment]
@robstoll Currently in a bit of time crunch, if you want to take this further it'll be great. Would not like to delay it further. I will look for other issues to contribute when I get some time.

**Comment by robstoll** [comment]
I opened a PR #730

**Comment by fthomas** [comment]
Fixed by #730.
